flink-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraConnectorITCase.java

```diff
@@ -438,6 +462,45 @@ public void cancel() {
 	}
 
 	@Test
+	public void testCassandraTableSink() throws Exception {
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		env.setParallelism(1);
+
+		DataStreamSource<Row> source = env.fromCollection(rowCollection);
+		CassandraTableSink cassandraTableSink = new CassandraTableSink(new ClusterBuilder() {
+			@Override
+			protected Cluster buildCluster(Cluster.Builder builder) {
+				return builder.addContactPointsWithPorts(new InetSocketAddress(HOST, PORT)).build();
+			}
+		}, INSERT_DATA_QUERY, new Properties());
+		CassandraTableSink newCassandrTableSink = cassandraTableSink.configure(FIELD_NAMES, FIELD_TYPES);
+
+		newCassandrTableSink.emitDataStream(source);
+
+		env.execute();
+		ResultSet rs = session.execute(SELECT_DATA_QUERY);
+		Assert.assertEquals(20, rs.all().size());
+	}
+
+	@Test
+	public void testCassandraTableSinkE2E() throws Exception {
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		env.setParallelism(4);
+		StreamTableEnvironment tEnv = StreamTableEnvironment.getTableEnvironment(env);
```

Comment:
OK, this does in fact require the t dependency on `flink-streaming-scala`. The `flink-scala` dependency isn't needed though.